lib/rust/frp2/src/node/class.rs

```diff
@@ -0,0 +1,111 @@
+use crate::prelude::*;
+
+use crate::data::Data;
+use crate::network::Network;
+use crate::runtime::with_runtime;
+use crate::runtime::EventConsumer;
+use crate::runtime::NodeData;
+use crate::runtime::NodeId;
+use crate::DefInfo;
+
+
+
+// ============
+// === Node ===
+// ============
+
+/// Any FRP node. This is a generalization for [`TypedNode`] with hidden `Type`.
+#[allow(missing_docs)]
+pub trait Node: Copy {
+    type Output: Data;
+    fn id(self) -> NodeId;
+}
+
+/// An alias for [`Node`] with default output type bounds.
+pub trait NodeWithDefaultOutput = Node where <Self as Node>::Output: Default;
+
+/// A generic node representation that is parametrized with the node kind and node output. All FRP
+/// nodes use this struct under the hood.
+#[derive(Derivative)]
+#[derivative(Copy(bound = ""))]
+#[derivative(Clone(bound = ""))]
+#[derivative(Debug(bound = ""))]
+#[repr(transparent)]
+pub struct TypedNode<Type, Output> {
+    pub(crate) _marker: PhantomData<*const (Type, Output)>,
+    pub(crate) id:      NodeId,
+}
+
+impl<Type, Output: Data> Node for TypedNode<Type, Output> {
+    type Output = Output;
+    #[inline(always)]
+    fn id(self) -> NodeId {
+        self.id
+    }
+}
+
+impl<Type, Output> TypedNode<Type, Output> {
+    /// # Safety
+    /// The value is checked to have the correct [`Output`] type.
+    #[inline(never)]
+    #[allow(unsafe_code)]
+    pub fn emit(&self, value: &Output)
+    where Output: Data {
+        with_runtime(|rt| unsafe { rt.unchecked_emit_borrow(self.id, value) });
+    }
+}
```

Comment:
Hmm, Hmm, this is harder than it looks to be. Checking them with debug assertions would require storing runtime type repo, so making `Data` use of `Any`, which is slower. This can be done with some magic. I added comment to consider that in my upcoming PR.